[WIP] Add Boot Logo to Sugar

vipulgupta2048 opened the pull request:
**Many more changes to Come**

This is still a work in progress to solve issue #6,
I request @walterbender @quozl to review but not merge the work done till now. 
Thanks.

vipulgupta2048 committed:
Add Sugar Labs plymouth splash and hook script

Sugar Labs plymouth theme is added for showing the boot logo. Contains plymouth script, media files and other components that are needed.

A hook script added for setting theme sugarlabs as default for Sugar.

vipulgupta2048 committed:
Add packages in package-lists for installing in live build

plymouth and plymouth-themes packages are added in the package-list (desktop.list.chroot)
Necessary to run and apply the plymouth theme in Sugar.

quozl commented:
Reviewed to 7a6ef7d.

* [ ] many files are being added twice or more; please work out exactly which files are needed and remove the files that are not needed; use tools such as `atime`, or split the patch set and use `git bisect` to minimise the test cycles,
* [ ] in the configuration files, where practical Sugar Labs should have a space between Sugar and Labs, and Labs should have an initial upper-case letter,
* [ ] the logos are fixed resolution, can you use scalable graphics instead, such as SVG?
* [ ] some of the logos are unrelated to the project, like the one for XUbuntu,
* [ ] the commit messages seem very vague and general; it isn't clear what you have done.

vipulgupta2048 commented:
Thanks for the review @quozl, implementing the changes now. 
- I am aware of the files added twice, will remove it in subsequent commits. It was just meant for testing purposes. I will surely check out the tools you mentioned. 
- For uniformity, I agree, I should go with Sugar Labs (Wherever practical) otherwise I can use sugarlabs for the name of the scripts.
- I have never tried SVG in Plymouth (It's really old, I am not sure it will support. **Update** It's not running. I would still be looking for workarounds. Used this in the [test](https://wiki.sugarlabs.org/go/File:Logo_alt_1.png)
- Didn't find the Xubuntu logo that you were referring to, but did remove other files.
- Commits from now on will be more clear, and descriptive.

vipulgupta2048 committed:
Remove /includes folder as per review

Duplicate files that were not necessary for the build were removed in this commit.
Files removed includes script.so and duplicate plymouth splash theme from includes directory.

vipulgupta2048 committed:
Delete files/images that were not needed as per review

vipulgupta2048 committed:
Modify Plymouth Script

Removed parts of the script that were not needed.

vipulgupta2048 committed:
Modify hook script as per review

- Add name of creator
- Fixed name 'Sugar Labs'
- Add PUBLIC DOMAIN

vipulgupta2048 committed:
Fix name "Sugar Labs" as per review in sugarlabs.plymouth

vipulgupta2048 commented:
- Updated my previous comment, please check the same.
- I hope the commits made are more clearer this time, please do provide more suggestions on improving.  
- [ ] Need to add a license to the script (did some major changes)  https://github.com/sugarlabs/sugar-live-build/pull/8/commits/b7f0a4ad42f49188934311f394320ee8968b7b30
@walterbender @quozl Suggestion appreciated. I say since it was GPL used for the script originally by Xubuntu, we should use a GPL too. What say? If yes, how to get it made?

walterbender commented:
I see no reason not to continue using the GPL

vipulgupta2048 commented:
Works with me, I should probably change the organisation name to Sugar Labs.

vipulgupta2048 committed:
Add Travis CI intergration for testing purposes

vipulgupta2048 committed:
Update .travis.yml

quozl commented:
Thanks.  Reviewed to 5eaf682.  Noted pull request title is "WIP".  Probably good progress, but I'm not sure.  Here are my concerns;

* [ ] you are adding the same file twice for no reason; `logo.png` and `logo_16bit.png` are the same file, and your `sugarlabs.script` file chooses one according to display bit depth; if the files are the same, then there's no reason to choose one; please simplify that part of the script and the redundant file,
* [ ] the spinner is not the same one as used on https://sugarlabs.org/, please work with the [web site repository](https://github.com/sugarlabs/www-sugarlabs),
* [ ] `sugarlabs.plymouth` has a line at end of file without a line break following,
* [ ] `0901-sugarplymouth.hook.chroot` has a `set -e`, why is that necessary?,
* [ ] `0901-sugarplymouth.hook.chroot` has an `update-initramfs`, very time consuming; why is that necessary, won't it happen anyway during kernel install?  Wasteful to do it more than needed.
* [ ] please revert 1cce205 and 5eaf682.  You do not have consensus to turn on continuous integration for this repository; it doesn't test much, and I don't think I need it,
* [ ] please rebase and squash the patches to simplify the change to as few commits as possible, and use a good commit message that explains what you are doing and why you are doing it, see [making commits](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits),

Also, have you tested this change?  You've not said if you have, in pull request comments or commit messages, nor given any screenshots.  I'd rather not merge untested work.

martinwguy commented:
Hi Vipul!
  I'm looking at maintaining the Sugar Live Build, and getting an appropriate boot logo for Grub and/or Debian splash screen in on my list of things I'd like to do. Are you happy with the current state of your fork so I can look at this?  Blessings -M
